src/scipp/signal/__init__.py

```diff
@@ -25,14 +26,31 @@ class SOS:
     coord: Variable
     sos: ndarray
 
+    def filtfilt(self, obj: Union[Variable, DataArray], dim: str,
+                 **kwargs) -> DataArray:
+        """
+        Forwards to :py:func:`scipp.signal.sosfiltfilt` with sos argument set to the SOS
+        instance.
+
+        The input data may be a variable or a data array. In the case of a variable, a
+        data array with the coord used for setting up the SOS parameters by
+        :py:func:`scipp.signal.butter` is created and forwarded to
+        :py:func:`scipp.signal.sosfiltfilt`.
+        """
+        if isinstance(obj, DataArray):
+            da = obj
+        else:
+            da = DataArray(data=obj, coords={dim: self.coord})
+        return sosfiltfilt(da=da, dim=dim, sos=self, **kwargs)
```

Comment:
I don't understand why this function has a different interface from the free function. Why not do `SOSO.filtfil = <turn-into-method>(sosfiltfilt)` (see how we bind methods to `Variable` for `<turn-into-method>`.)
It would be unnecessarily confusing if both functions take different argument types.

Reply:
Fixed.